audformat/core/table.py

```diff
@@ -205,6 +205,79 @@ def drop_columns(
 
         return self
 
+    def drop_index(
+            self,
+            index: pd.Index,
+            *,
+            inplace: bool = False,
+    ) -> 'Base':
```

Comment:
I guess we need something like

```suggestion
    ) -> typing.Union['Base', 'Table', 'MiscTable']:
```

or maybe

```suggestion
    ) -> typing.Union['Table', 'MiscTable']:
```

as `'Base'` is not part of the official API.
I also don't know, if you can combine strings like `'Table'` with `typing.Union`

Reply:
Ah good point. I will also change it with the other functions.

Reply:
Or maybe there is even a way to tell it that it returns as type the one of `self`?

Reply:
For `copy()` we have solved in a not so nice why by adding the method also to `Table` and `MiscTable` with the correct typing and docstring.

Reply:
They will officially `typing.Self` add it in `3.11`, see https://peps.python.org/pep-0673/. 

With `from __future__ import annotations` I can use it without error, but unfortunately it's then no longer displaying the type in Sphinx, e.g.:

![image](https://user-images.githubusercontent.com/10383417/180795129-31c539b3-cb6b-4ea5-bb02-ae361d501bca.png)

is rendered as:

![image](https://user-images.githubusercontent.com/10383417/180795195-e4c79d20-8ec0-4f66-bc3b-0cbd21f21461.png)

So we either ignore that and hope it will show up once it's supported by Sphinx or we use `typing.Union[Table, MiscTable]` as you suggested earlier.

Reply:
We are using the [sphinx-autodoc-typehints](https://github.com/tox-dev/sphinx-autodoc-typehints) plugin at the moment, but it turns out that newer versions of Sphinx do support the same as well.

With this we could get something like

![image](https://user-images.githubusercontent.com/173624/180804944-b294b998-97f3-44c3-a921-829bd205a43d.png)

I will do a few more investigations if there are any unwanted consequences, but otherwise we might think about switching to no longer use the extension. But this I would do in another pull request.

Reply:
I also still want to figure out, why for `audformat` the links to the `pandas` type hints (like `pd.Index`) do not work, but for `audinterface` they do.

Reply:
Ok, then I would propose we stay with `typing.Self` here.

Reply:
I tested further using the build-in typing support of Sphinx. The big problem is that it is not able to add any custom HTML span to the text, but just marks it with `em`, which means we have no way to style it with CSS and cannot adjust it to look like we want it to look. Which means I would not recommend to use it for now.

I would suggest that we stay with `typing.Self` here and not having the return type documented in the docs. Maybe, we can open a pull request in `sphinx-autodoc-typehints` to support it, or open an issue that it would be nice if `sphinx` could add span to the type hints when not used in the function signature with `autodoc_typehints = 'description'` (when used in the function signature it adds already spans to it).

Reply:
Ok

Reply:
I created https://github.com/sphinx-doc/sphinx/issues/10711